Compass/ViewController.swift

```diff
@@ -51,6 +48,59 @@ class ViewController: UIViewController, CLLocationManagerDelegate {
     override func viewDidLoad() {
         super.viewDidLoad()
         initLocationManager()
+        createUI()
+    }
+    
+    func createUI(){
+        
+        navigationController?.navigationBar.barTintColor = UIColor.black
+        self.view.backgroundColor = UIColor.darkGray
+        self.navigationController?.navigationBar.titleTextAttributes = [NSAttributedStringKey.foregroundColor: UIColor.green]
+        navigationItem.title = "No storyboard"
+        degreeLabel.text = "Dir Degree->"
+        self.view.addSubview(degreeLabel)
+        self.view.addSubview(ArrowImage)
+        
+        setTargetLocationButton.setTitle("Set Target Location", for: .normal)
+        setTargetLocationButton.addTarget(self, action: #selector(setTargetLocationPressed), for: .touchUpInside)
+        self.view.addSubview(setTargetLocationButton)
+        
+        setTargetLocationButton.translatesAutoresizingMaskIntoConstraints = false
+        NSLayoutConstraint.activate([ setTargetLocationButton.centerXAnchor.constraint(lessThanOrEqualTo:     self.view.centerXAnchor),
+                                      setTargetLocationButton.centerYAnchor.constraint(equalTo: self.view.centerYAnchor, constant: 140)])
+        
+        degreeLabel.translatesAutoresizingMaskIntoConstraints = false
+        NSLayoutConstraint.activate([ degreeLabel.centerXAnchor.constraint(lessThanOrEqualTo:self.view.centerXAnchor),
+                                      degreeLabel.centerYAnchor.constraint(equalTo: self.view.centerYAnchor, constant: 100)])
+        
+        let backButton = UIButton(frame: CGRect.zero)
+        backButton.setTitle("Back", for: .normal)
+        backButton.addTarget(self, action: #selector(backButtonPressed), for: .touchUpInside)
```

Comment:
Sometimes this createUI method can get quite long, so one thing we do is, at the top of our view or view controller we separate out the view set up into properties. So at the top of the class we might have something like this:
```
private lazy var emailButton: ROOButton = {
        let button = UIButton(frame: CGRect.zero)
        button.translatesAutoresizingMaskIntoConstraints = false
        button.setTitle("Back", for: .normal)
        return button
    }()
```
The lazy means the code isn't run until the button is actually accessed, which can be much more efficient if you have lots of properties that might never be accessed under certain conditions.

In this case there's probably not much to be gained from this, but sometimes we have buttons with a lot of setup code (e.g. Here's one I pulled from the iOS client:
```
private lazy var emailButton: ROOButton = {
        let button = ROOButton.init(style: .primary)
        button.translatesAutoresizingMaskIntoConstraints = false
        button.rooText = NSLocalizedString("signUpOrLogIn.emailButton", comment: "email button text")
        let icon = Icon.envelope.rooResizedImage(to: CGSize(width: 18, height: 18))
        button.setImage(icon, for: .normal)
        button.imageEdgeInsets = UIEdgeInsets(top: 0, left: 0, bottom: 0, right: 8.0)
        button.tintColor = Color.white
        return button
    }()
```